From 4f9d99447f2936c0520bc3a0baedb27cb7eb608d Mon Sep 17 00:00:00 2001
From: Wen He <wen.he_1@nxp.com>
Date: Thu, 22 Aug 2019 11:08:22 +0800
Subject: [PATCH 585/741] drm/imx/hdp: Remove the simple pixel clock framework

commit b4af87d94af1549afd8963adf0a81973a6ca48a2 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Commit e2e02dda4754 ("clk: ls1028a: Add clock driver for Display
output interface") use the clock framework of linux instead of the
simple with ls1028a_pixel_link_mux(), so remove these code.

Signed-off-by: Wen He <wen.he_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c | 41 +------------------------------
 1 file changed, 1 insertion(+), 40 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index f41a29c8d2d9..aff5397d1957 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -72,36 +72,6 @@ static void imx_hdp_state_init(struct imx_hdp *hdp)
 	state->edp = hdp->is_edp;
 }
 
-static void ls1028a_pixel_link_mux(state_struct *state,
-				   const struct drm_display_mode *mode)
-{
-	struct imx_hdp *hdp = state_to_imx_hdp(state);
-	u32 val;
-
-	if (mode->hdisplay == 3840
-	    && mode->vdisplay == 2160)
-		val = 0x0402002c;
-	else if (mode->hdisplay == 1920
-		 && mode->vdisplay == 1080)
-		val = 0x1002002c;
-	else if (mode->hdisplay == 1280
-		 && mode->vdisplay == 720)
-		val = 0x2002002c;
-	else if (mode->hdisplay == 720
-		 && mode->vdisplay == 480)
-		val = 0x5802002c;
-	else
-		/* Set deafault pixel clock to 1080p*/
-		val = 0x1002002c;
-
-	writel(val, hdp->mem.ss_base + CSR_PLLDIG_PLLDV);
-
-	/* Recommend register set for PLL Divider */
-	writel(0x40000000, hdp->mem.ss_base + CSR_PLLDIG_PLLFM);
-	writel(0x40030000, hdp->mem.ss_base + CSR_PLLDIG_PLLFD);
-	writel(0x44000000, hdp->mem.ss_base + CSR_PLLDIG_PLLCAL1);
-	writel(0x0005002b, hdp->mem.ss_base + CSR_PLLDIG_PLLCAL2);
-}
 static void imx8qm_pixel_link_mux(state_struct *state,
 				  const struct drm_display_mode *mode)
 {
@@ -1232,7 +1202,6 @@ static struct hdp_ops ls1028a_dp_ops = {
 	.get_edid_block = dp_get_edid_block,
 	.get_hpd_state = dp_get_hpd_state,
 	.phy_reset = ls1028a_phy_reset,
-	.pixel_link_mux = ls1028a_pixel_link_mux,
 	.clock_init = ls1028a_clock_init,
 	.ipg_clock_enable = ls1028a_ipg_clock_enable,
 	.ipg_clock_disable = ls1028a_ipg_clock_disable,
@@ -1445,16 +1414,8 @@ static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 		dev_warn(dev, "Failed to get HDP RESET base register\n");
 #else
 
-	/* multimeida register map */
-	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-	hdp->mem.ss_base = devm_ioremap_resource(dev, res);
-	if (IS_ERR(hdp->mem.ss_base)) {
-		dev_err(dev, "Failed to get Multimedia PLL base register\n");
-		return -EINVAL;
-	}
-
 	/* register map */
-	res = platform_get_resource(pdev, IORESOURCE_MEM, 1);
+	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	hdp->mem.regs_base = devm_ioremap_resource(dev, res);
 	if (IS_ERR(hdp->mem.regs_base)) {
 		dev_err(dev, "Failed to get HDP CTRL base register\n");
-- 
2.17.1

